pkg/rewards/2_goldStakerRewardAmounts.go

```diff
@@ -107,26 +108,29 @@ staker_operator_total_tokens AS (
     END as total_staker_operator_payout
   FROM staker_proportion
 ),
--- Calculate the token breakdown for each (staker, operator) pair
+-- Calculate the token breakdown for each (staker, operator) pair with dynamic split logic
+-- If no split is found, default to 1000 (10%)
 token_breakdowns AS (
-  SELECT *,
+  SELECT sott.*,
     CASE
-      WHEN snapshot < @amazonHardforkDate AND reward_submission_date < @amazonHardforkDate THEN
-        cast(total_staker_operator_payout * 0.10 AS DECIMAL(38,0))
-      WHEN snapshot < @nileHardforkDate AND reward_submission_date < @nileHardforkDate THEN
-        (total_staker_operator_payout * 0.10)::text::decimal(38,0)
+      WHEN sott.snapshot < @amazonHardforkDate AND sott.reward_submission_date < @amazonHardforkDate THEN
+        cast(sott.total_staker_operator_payout * 0.10 AS DECIMAL(38,0))
+      WHEN sott.snapshot < @nileHardforkDate AND sott.reward_submission_date < @nileHardforkDate THEN
+        (sott.total_staker_operator_payout * 0.10)::text::decimal(38,0)
       ELSE
-        floor(total_staker_operator_payout * 0.10)
+        floor(sott.total_staker_operator_payout * COALESCE(oas.split, 1000) / CAST(10000 AS DECIMAL))
```

Comment:
I don't think this cast is actually doing anything? Since we already know 10000 will be a decimal by declaring it as a constant. My previous comment was moreso just sanity checking the types more than anything else

Reply:
Oh, do you mean to cast the output of `split/10000` ?

Reply:
This casting ensures that the resulting division will be of type `DECIMAL`. 

1. When `oas.split` is `NULL`:

`COALESCE(oas.split, 1000)` returns `1000` (an `INTEGER`).
`1000 / CAST(10000 AS DECIMAL)` results in `0.10` as `DECIMAL`, maintaining the same behavior as `0.10` prior to the change.

2. When `oas.split` is not `NULL`:

The result will still be computed with `DECIMAL` precision, ensuring no loss of precision during the division.

Reply:
Also suggested by @seanmcgary here: https://github.com/Layr-Labs/sidecar/pull/117#discussion_r1856795893

Reply:
i.e `INTEGER` / `DECIMAL` in Postgres ensures that the resulting value is of type `DECIMAL`.

Reply:
Also the reason to do this way vs cast `split/10000` is so that precision is maintained. In the former we multiply by numerator first and then divde by denominator. Vs for the latter we do division first leading to precision loss (albeit not too much).

Reply:
Okay, all for the explicitness here